src/cai_lab_to_nwb/zaki_2024/imaginginterface.py

```diff
@@ -0,0 +1,309 @@
+
+        self.photon_series_type = "OnePhotonSeries"
+
+    def _get_session_start_time(self):
+
+        general_metadata_json = self.folder_path / "metaData.json"
+        assert general_metadata_json.exists(), f"General metadata json not found in {self.folder_path}"
+
+        ## Read metadata
+        with open(general_metadata_json) as f:
+            general_metadata = json.load(f)
+
+        session_start_time = datetime.datetime(
+            year=general_metadata["year"],
+            month=general_metadata["month"],
+            day=general_metadata["day"],
+            hour=general_metadata["hour"],
+            minute=general_metadata["minute"],
+            second=general_metadata["second"],
+            microsecond=general_metadata["msec"] * 1000,  # Convert milliseconds to microseconds
+        )
+
+        return session_start_time
+
+    def get_metadata(self) -> DeepDict:
+        from neuroconv.tools.roiextractors import get_nwb_imaging_metadata
+
+        metadata = super().get_metadata()
+        default_metadata = get_nwb_imaging_metadata(self.imaging_extractor, photon_series_type=self.photon_series_type)
+        metadata = dict_deep_update(metadata, default_metadata)
+        metadata["Ophys"].pop("TwoPhotonSeries", None)
+
+        session_start_time = self._get_session_start_time()
+        metadata["NWBFile"].update(session_start_time=session_start_time)
+
+        device_metadata = metadata["Ophys"]["Device"][0]
+        miniscope_config = deepcopy(self._miniscope_config)
+        device_name = miniscope_config.pop("name")
+        device_metadata.update(name=device_name, **miniscope_config)
+        # Add link to Device for ImagingPlane
+        imaging_plane_metadata = metadata["Ophys"]["ImagingPlane"][0]
+        imaging_plane_metadata.update(
+            device=device_name,
+            imaging_rate=self.imaging_extractor.get_sampling_frequency(),
+        )
+        one_photon_series_metadata = metadata["Ophys"]["OnePhotonSeries"][0]
+        one_photon_series_metadata.update(unit="px")
+
+        return metadata
+
+    def get_metadata_schema(self) -> dict:
+        metadata_schema = super().get_metadata_schema()
+        metadata_schema["properties"]["Ophys"]["definitions"]["Device"]["additionalProperties"] = True
+        return metadata_schema
+
+    def get_original_timestamps(self) -> np.ndarray:
+
+        timestamps_file_path = self.folder_path / "Miniscope" / "timeStamps.csv"
+        assert timestamps_file_path.exists(), f"Miniscope timestamps file not found in {self.folder_path}"
+        
+        import pandas as pd 
+        
+        timetsamps_df = pd.read_csv(timestamps_file_path)  
+        timestamps_milliseconds = timetsamps_df["Time Stamp (ms)"].values.astype(float)
+        timestamps_seconds = timestamps_milliseconds / 1000.0
+        
+        # Shift when the first timestamp is negative
+        # TODO: Figure why, I copied from miniscope
```

Comment:
I guess the timestamps refer to the session start time. If this is correct, when shifting all timestamps, we should also shift back the session start time.

Reply:
Let's discuss this briefly in the meeting. I copied this from the current miniscope code on the corresponding extension. I would rather fix the timing issues in one go when we are at the synch stage.